Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -678,30 +688,32 @@ void DirectoryBlobReader::SetPartitions(std::vector<PartitionWithType>&& partiti
   }
   m_data_size = partition_address;
 
-  m_nonpartition_contents.AddReference(PARTITION_TABLE_ADDRESS, m_partition_table);
+  m_nonpartition_contents.Add(PARTITION_TABLE_ADDRESS, std::move(partition_table));
 }
 
 // This function sets the header that's shortly before the start of the encrypted
 // area, not the header that's right at the beginning of the encrypted area
 void DirectoryBlobReader::SetPartitionHeader(DirectoryBlobPartition* partition,
-                                             u64 partition_address)
+                                             u64 partition_address,
+                                             std::unique_ptr<DiscIO::VolumeDisc> volume)
```

Comment:
The interface would be cleaner without `std::unique_ptr`:

```suggestion
                                             DiscIO::VolumeDisc* volume)
```